pkg/aws/aws_client/role.go

```diff
@@ -89,3 +96,202 @@ func (client *AWSClient) ListRoles() ([]types.Role, error) {
+	if err != nil {
+		fmt.Println("Failed to convert Role Policy Document into JSON: ", err)
+		return types.Role{}, err
+	}
+	return client.CreateRole(roleName, assumeRolePolicyDocument, "", make(map[string]string), "/")
+}
+
+func (client *AWSClient) CreateRoleForAuditLogForward(roleName, awsAccountID string, oidcEndpointURL string) (types.Role, error) {
+	statement := map[string]interface{}{
+		"Effect": "Allow",
+		"Principal": map[string]interface{}{
+			"Federated": fmt.Sprintf("arn:aws:iam::%s:oidc-provider/%s", awsAccountID, oidcEndpointURL),
+		},
+		"Action": "sts:AssumeRoleWithWebIdentity",
+		"Condition": map[string]interface{}{
+			"StringEquals": map[string]interface{}{
+				fmt.Sprintf("%s:sub", oidcEndpointURL): "system:serviceaccount:openshift-config-managed:cloudwatch-audit-exporter",
+			},
+		},
+	}
+
+	assumeRolePolicyDocument, err := completeRolePolicyDocument(statement)
+	if err != nil {
+		fmt.Println("Failed to convert Role Policy Document into JSON: ", err)
+		return types.Role{}, err
+	}
+
+	return client.CreateRole(roleName, string(assumeRolePolicyDocument), "", make(map[string]string), "/")
+}
+
+func (client *AWSClient) CreatePolicy(policyName string, statements ...map[string]interface{}) (string, error) {
+	timeCreation := time.Now().Local().String()
+	description := fmt.Sprintf("Created by OCM QE at %s", timeCreation)
+	document := map[string]interface{}{
+		"Version":   "2012-10-17",
+		"Statement": []map[string]interface{}{},
+	}
+	if len(statements) != 0 {
+		for _, statement := range statements {
+			document["Statement"] = append(document["Statement"].([]map[string]interface{}), statement)
+		}
+	}
+	documentBytes, err := json.Marshal(document)
+	if err != nil {
+		err = fmt.Errorf("Error to unmarshal the statement to string, %v", err)
+	}
+	documentStr := string(documentBytes)
+	policyCreator := iam.CreatePolicyInput{
+		PolicyDocument: &documentStr,
+		PolicyName:     &policyName,
+		Description:    &description,
+	}
+	outRes, err := client.IamClient.CreatePolicy(context.TODO(), &policyCreator)
+	if err != nil {
+		return "", err
+	}
+	policyArn := *outRes.Policy.Arn
+	return policyArn, err
+}
+
+func (client *AWSClient) CreatePolicyForAuditLogForward(policyName string) (string, error) {
+
+	statement := map[string]interface{}{
+		"Effect":   "Allow",
+		"Resource": "arn:aws:logs:*:*:*",
+		"Action": []string{
+			"logs:PutLogEvents",
+			"logs:CreateLogGroup",
+			"logs:PutRetentionPolicy",
+			"logs:CreateLogStream",
+			"logs:DescribeLogGroups",
+			"logs:DescribeLogStreams",
+		},
+	}
+	return client.CreatePolicy(policyName, statement)
+}
+
+func completeRolePolicyDocument(statement map[string]interface{}) (string, error) {
+	rolePolicyDocument := map[string]interface{}{
+		"Version":   "2012-10-17",
+		"Statement": statement,
+	}
+
+	assumeRolePolicyDocument, err := json.Marshal(rolePolicyDocument)
+	if err != nil {
+		return "", err
+	}
+	return string(assumeRolePolicyDocument), nil
```

Comment:
i think if there's an error returned by the json function above, we can remove the above if and return and just return err here